elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/PredicateAnalyzer.java

```diff
@@ -873,8 +866,37 @@ private SimpleQueryExpression(NamedFieldExpression rel) {
       builder = boolQuery().mustNot(termsQuery(getFieldReference(), iterable));
       return this;
     }
-  }
 
+    @Override public QueryExpression range(LiteralExpression literal) {
+      Iterator<?> iterator = ((Iterable<?>) literal.value()).iterator();
+      BoolQueryBuilder boolQuery = boolQuery();
+      while (iterator.hasNext()) {
+        Range range = (Range) iterator.next();
```

Comment:
I don't have much context, but usually I would assume it should be some RangeExpression, or at least some utils to extract the Range expressions, instead of expose the iterator directly here?

would that be better ?

Reply:
Originally I would like to add a class 'RangeExpression' to convey the information of 'Range Set'.  When i began to implement it, i found the 'RangeExpression''s structure will be almost same with the class 'Range'.  The benefit to add a new RangeExpression looks not that much. So i just directly return a list of 'Range' and build the rangeQueryBuilder for each Range.  However you are right and I should at least move the extraction and transformation of the Range to a kind of utility function.